document-store/src/main/java/org/hypertrace/core/documentstore/TypesafeDatastoreConfigAdapter.java

```diff
@@ -19,43 +19,14 @@ interface TypesafeDatastoreConfigAdapter {
   class MongoTypesafeDatastoreConfigAdapter implements TypesafeDatastoreConfigAdapter {
     @Override
     public DatastoreConfig convert(final Config config) {
-      final MongoConnectionConfig connectionConfig =
-          (MongoConnectionConfig)
-              TypesafeConfigDatastoreConfigExtractor.from(config, DatabaseType.MONGO)
-                  .hostKey("host")
-                  .portKey("port")
-                  .usernameKey("user")
-                  .passwordKey("password")
-                  .databaseKey("database")
-                  .authDatabaseKey("authDatabase")
-                  .applicationNameKey("applicationName")
-                  .extract()
-                  .connectionConfig();
-
       final MongoConnectionConfig overridingConnectionConfig =
-          new MongoConnectionConfig(
-              connectionConfig.endpoints(),
-              connectionConfig.database(),
-              connectionConfig.credentials(),
-              connectionConfig.applicationName(),
-              null) {
+          new MongoConnectionConfig(emptyList(), null, null, "", null) {
```

Comment:
So, now it will work with old config if connectionUrl is provided, right?

Reply:
Yes.